packages/devtools_server/lib/src/chrome.dart

```diff
@@ -0,0 +1,146 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:webkit_inspection_protocol/webkit_inspection_protocol.dart';
+
+// TODO(kenzie): move this code to dart-lang/browser_launcher. This code was
+// copied from https://github.com/dart-lang/webdev/blob/master/webdev/lib/src/serve/chrome.dart
+
+const _chromeEnvironment = 'CHROME_EXECUTABLE';
+const _linuxExecutable = 'google-chrome';
+const _macOSExecutable =
+    '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome';
+const _windowsExecutable = r'Google\Chrome\Application\chrome.exe';
+
+String get _executable {
+  if (Platform.environment.containsKey(_chromeEnvironment)) {
+    return Platform.environment[_chromeEnvironment];
+  }
+  if (Platform.isLinux) return _linuxExecutable;
+  if (Platform.isMacOS) return _macOSExecutable;
+  if (Platform.isWindows) return _windowsExecutable;
+  throw StateError('Unexpected platform type.');
+}
+
+var _currentCompleter = Completer<Chrome>();
+
+/// A class for managing an instance of Chrome.
+class Chrome {
+  final int debugPort;
+  final Process _process;
+  final Directory _dataDir;
+
+  final ChromeConnection chromeConnection;
+
+  Chrome._(
+      this.debugPort,
+      this.chromeConnection, {
+        Process process,
+        Directory dataDir,
+      })  : _process = process,
+        _dataDir = dataDir;
+
+  Future<void> close() async {
+    if (_currentCompleter.isCompleted) _currentCompleter = Completer<Chrome>();
+    chromeConnection.close();
+    _process?.kill();
+    await _process?.exitCode;
+    await _dataDir?.delete(recursive: true);
+  }
+
+  /// Connects to an instance of Chrome with an open debug port.
+  static Future<Chrome> fromExisting(int port) async =>
+      _connect(Chrome._(port, ChromeConnection('localhost', port)));
+
+  static Future<Chrome> get connectedInstance => _currentCompleter.future;
+
+  /// Starts Chrome with the remote debug port enabled.
+  ///
+  /// Each url in [urls] will be loaded in a separate tab.
+  static Future<Chrome> start(List<String> urls, {int port}) async {
+    var dataDir = Directory.systemTemp.createTempSync();
+    port = port == null || port == 0 ? await findUnusedPort() : port;
+    var args = [
```

Comment:
pass the args in rather than setting them here. we should not be using any of these args.